controllers/managedocs_controller.go

```diff
@@ -633,14 +647,12 @@ func (r *ManagedOCSReconciler) reconcileStorageCluster() error {
 }
 
 func (r *ManagedOCSReconciler) getDesiredConvergedStorageCluster() (*ocsv1.StorageCluster, error) {
-	addonParams := r.addonParamSecret.Data
-
-	sizeAsString := string(addonParams[storageClassSizeKey])
-	enableMCGRaw, exists := addonParams[enableMCGKey]
+	sizeAsString := r.addonParams[storageClassSizeKey]
+	enableMCGRaw, exists := r.addonParams[enableMCGKey]
 	// Setting hardcoded value here to force no MCG deployment
 	enableMCGAsString := "false"
 	if exists {
-		enableMCGAsString = string(enableMCGRaw)
+		enableMCGAsString = enableMCGRaw
 	}
```

Comment:
This can be refactored into a form that does not pollute the function scope with an `exists` var 
```
	sizeAsString := r.addonParams[storageClassSizeKey]
	
	// Setting hardcoded value here to force no MCG deployment
	enableMCGAsString := "false"
	if enableMCGRaw, exists := r.addonParams[enableMCGKey]; exists {
		enableMCGAsString = enableMCGRaw
	}
```